### PR TITLE
Pass throught custom http headers as transport configuration

### PR DESCRIPTION
**Pull request by fungs**
This minimalistic PR adds the possibility to specify custom http headers as transport configuration.

Example:

```
component = AutobahnComponent(
  transports={
    "type": "websocket",
    "url": "YOUR URL",
    "headers": {"Authorization": "YOUR CUSTOM AUTHORIZATION"},
  },
    ...
)
```

The patch does not change the internal logic, the configuration is automatically passed and used in the websocket connection factory. I tried to keep the changes minimal (although I was tempted to apply the boy scout rule and do some tidy up work).

### CODE REVIEW

**Review by oberstet — CHANGES_REQUESTED** [review]
thanks a lot! I like the "minimalistic" approach. +1 for the additional check that @meejah mentioned as this is a transport-specific feature while "component" is exactly about abstracting away "all" transport details (like in general with WAMP at the app level, above transports). for me, a simple/deadly "assert" would be fine .. that's enough to make it fail and the user/developer aware.

**Comment by oberstet** [comment]
rgd 

https://github.com/crossbario/autobahn-python/actions/runs/8187466713/job/22419880678?pr=1630#step:8:116

pls see

https://github.com/crossbario/autobahn-python/pull/1628#issuecomment-1976024550

TLDR: this is fallout from a breaking change in pytest-asyncio ...

**Comment by fungs** [comment]
I will do the polishing next week.

**Comment by fungs** [comment]
> I will do the polishing next week.

Here we go, I raised a ValueError, as for the other incompatible arguments and removed the default None return value from get().

**Comment by oberstet** [comment]
flake is complaining about WS https://github.com/crossbario/autobahn-python/actions/runs/8357659564/job/22881485753?pr=1630#step:6:71

the asyncio issue is related to the latest (breaking) changes in pytest-asyncio rgd event loop setup I guess ..

File: autobahn/wamp/component.py [hunk 1]
```diff
@@ -229,6 +229,8 @@ def _create_transport(index, transport, check_native_endpoint=None):
             endpoint_config = transport['endpoint']
         if 'serializers' in transport:
             raise ValueError("'serializers' is only for websocket; use 'serializer'")
+        if headers is None:
```
Comment (meejah) [inline autobahn/wamp/component.py:232]: `is not None` I think..?

Comment (fungs) [reply autobahn/wamp/component.py]: should be fixed

**Comment by oberstet** [comment]
```
  autobahn/wamp/component.py:124:9: E123 closing bracket does not match indentation of opening bracket's line
  autobahn/wamp/component.py:163:1: W293 blank line contains whitespace
  1     E123 closing bracket does not match indentation of opening bracket's line
  1     W293 blank line contains whitespace
  flake8: exit 1 (4.27 seconds) /home/runner/work/autobahn-python/autobahn-python> flake8 -v --statistics --ignore=E402,E501,E722,E741,N801,N802,N803,N805,N806,N815,N818 --exclude 'autobahn/wamp/message_fbs.py,autobahn/wamp/gen/*' autobahn pid=2865
```

**Comment by oberstet** [comment]
rebased to master and rerunning CI .. let's see

https://github.com/crossbario/autobahn-python/pull/1638

**Comment by oberstet** [comment]
merged via #1638 - thanks for contributing!
